llama.cpp

```diff
@@ -212,6 +212,7 @@ enum llm_arch {
     LLM_ARCH_MINICPM,
     LLM_ARCH_GEMMA,
     LLM_ARCH_UNKNOWN,
+    LLM_ARCH_STARCODER2
```

Comment:
`LLM_ARCH_UNKNOWN` should stay the last.

Reply:
Thank you! This helped resolve the `segmentation fault: core dumped` error. Out of curiosity, why this leads to that error?

Reply:
That's a good question. I've investigated, and it occurs when trying to find the model architecture from it's name (in `llm_arch_from_string`). When it gets to `LLM_ARCH_UNKNOWN`, its name is a pointer to NULL (maybe by default a map with enum keys is initialized for all enums?) and so it segfaults.

I've tested other ways to make this happen, and it seems like any unknown name causes this, even when `LLM_ARCH_UNKNOWN` is last. I think I'll make a PR to fix this.